crates/gazette/src/lib.rs

```diff
@@ -35,6 +35,8 @@ pub enum Error {
     },
     #[error("{0}")]
     Protocol(&'static str),
+    #[error("{0}")]
+    Append(String),
```

Comment:
use `Protocol` for fixed string constants, or a more-specific error type wrapping `std::io::Error` for other errors as needed (similar to `ReadFragment` for example)